src/RageTimer.cpp

```diff
@@ -88,19 +82,19 @@ float RageTimer::GetDeltaTime()
 	return diff;
 }
 
-/*
- * Get a timer representing half of the time ago as this one.  This is
- * useful for averaging time.  For example,
- *
- * RageTimer tm;
- * ... do stuff ...
- * RageTimer AverageTime = tm.Half();
- * printf( "Something happened approximately %f seconds ago.\n", tm.Ago() );
- */
+/* Get a timer representing half of the time ago as this one.  This is	
+ * useful for averaging time.  For example,	
+ *	
+ * RageTimer tm;	
+ * ... do stuff ...	
+ * RageTimer AverageTime = tm.Half();	
+ * printf( "Something happened approximately %f seconds ago.\n", tm.Ago() ); 
+ * Note this has been reverted to the original SM3.95 function. */
 RageTimer RageTimer::Half() const
 {
-	const float fProbableDelay = Ago() / 2;
-	return *this + fProbableDelay;
+	const RageTimer now;
+	const float ProbableDelay = -(now - *this) / 2;
```

Comment:
I was reading this from the 3.95 codebase and that codebase is actually wrong.
Ago() / 2 is correct. You can alternatively remove the negative sign but it's functionally equivalent.

You can do some math here to verify:

say `*this` reports 10ms. 10ms later, `now` will report 20ms. -(20-10) / 2 = -5ms.
`*this + ProbablyDelay` would actually be 10 + (-5) which is 5ms instead of 15ms as expected.